dev/DynamicDependency/MddBootstrap.cpp

```diff
@@ -0,0 +1,378 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+
+#include "MddBootstrap.h"
+#include "MddBootstrapTest.h"
+
+#include "MsixDynamicDependency.h"
+
+#include "IDynamicDependencyLifetimeManager.h"
+
+wil::unique_cotaskmem_ptr<BYTE[]> GetFrameworkPackageInfoForPackage(PCWSTR packageFullName, const PACKAGE_INFO*& frameworkPackageInfo);
+DLL_DIRECTORY_COOKIE AddFrameworkToPath(PCWSTR path);
+void RemoveFrameworkFromPath(PCWSTR frameworkPath);
+CLSID FindDDLM(const PACKAGE_VERSION minVersion);
+CLSID GetClsid(const winrt::Windows::ApplicationModel::AppExtensions::AppExtension appExtension);
+
+IDynamicDependencyLifetimeManager* g_lifetimeManager{};
+wil::unique_hmodule g_projectReunionDll{};
+wil::unique_process_heap_string g_packageDependencyId;
+MDD_PACKAGEDEPENDENCY_CONTEXT g_packageDependencyContext{};
+
+static std::wstring g_test_ddlmPackageNamePrefix;
+static std::wstring g_test_ddlmPackagePublisherId;
+
+inline winrt::Windows::System::ProcessorArchitecture GetCurrentArchitecture()
+{
+#if defined(_M_X64)
+    return winrt::Windows::System::ProcessorArchitecture::X64;
+#elif defined(_M_IX86)
+    return winrt::Windows::System::ProcessorArchitecture::X86;
+#elif defined(_M_ARM64)
+    return winrt::Windows::System::ProcessorArchitecture::Arm64;
+#elif defined(_M_ARM)
+    return winrt::Windows::System::ProcessorArchitecture::Arm;
+#else
+#   error "Unknown processor architecture"
+#endif
+}
+
+inline PCWSTR GetCurrentArchitectureAsString()
+{
+#if defined(_M_X64)
+    return L"x64";
+#elif defined(_M_IX86)
+    return L"x86";
+#elif defined(_M_ARM64)
+    return L"arm64";
+#elif defined(_M_ARM)
+    return L"arm";
+#else
+#   error "Unknown processor architecture"
+#endif
+}
+
+inline winrt::Windows::System::ProcessorArchitecture ParseArchitecture(PCWSTR architecture)
+{
+    if (CompareStringOrdinal(architecture, -1, L"x64", -1, TRUE) == CSTR_EQUAL)
+    {
+        return winrt::Windows::System::ProcessorArchitecture::X64;
+    }
+    else if (CompareStringOrdinal(architecture, -1, L"x86", -1, TRUE) == CSTR_EQUAL)
+    {
+        return winrt::Windows::System::ProcessorArchitecture::X86;
+    }
+    else if (CompareStringOrdinal(architecture, -1, L"arm64", -1, TRUE) == CSTR_EQUAL)
+    {
+    return winrt::Windows::System::ProcessorArchitecture::Arm64;
```

Comment:
Something happened to the indentation here. (cc @chrisguzak for that `.clang-format` suggestion)

Reply:
Odd. Thanks